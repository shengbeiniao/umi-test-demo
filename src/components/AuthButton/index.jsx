import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Button } from 'antd'

export default class AuthButton extends PureComponent {
  static propTyps = {
    style: PropTypes.object,
    role: PropTypes.oneOf(['admin', 'user']),
    children: PropTypes.string,
    onClick: PropTypes.func
  }

  constructor(props) {
    super(props)
    this.state = { isClicked: false }
  }

  onClick = () => {
    setTimeout(()=>{
      this.setState({ isClicked: true })
    },1000)
    this.props.onClick && this.props.onClick()
  }

  render() {
    const { style, role, children } = this.props
    return (
      <Button
        style={style}
        disabled={role !== 'admin'}
        onClick={this.onClick}
        type={this.state.isClicked ? 'primary' : 'common'}>
        {children}
      </Button>
    )
  }
}
