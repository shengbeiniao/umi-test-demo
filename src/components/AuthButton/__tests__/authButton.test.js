import { act, create } from 'react-test-renderer'
import AuthButton from '../index.jsx'

jest.useFakeTimers()
test('role/admin', () => {
  let root = create(<AuthButton role="admin">Test</AuthButton>)
  expect(root.toJSON()).toMatchSnapshot()
  act(() => {
    root.getInstance().onClick()
    jest.runAllTimers()
  })
  expect(root.toJSON()).toMatchSnapshot()
})

test('role/user', () => {
  let root = create(<AuthButton role="user">Test</AuthButton>)
  expect(root.toJSON()).toMatchSnapshot()
})
