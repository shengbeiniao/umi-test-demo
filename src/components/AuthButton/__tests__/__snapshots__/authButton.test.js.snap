// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`role/admin 1`] = `
<button
  className="ant-btn ant-btn-common"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span>
    Test
  </span>
</button>
`;

exports[`role/admin 2`] = `
<button
  className="ant-btn ant-btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span>
    Test
  </span>
</button>
`;

exports[`role/user 1`] = `
<button
  className="ant-btn ant-btn-common"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span>
    Test
  </span>
</button>
`;
